source/common/http/conn_pool_grid.cc

```diff
@@ -311,10 +310,11 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
   // New streams should not be created during draining.
   ASSERT(!draining_);
 
-  if (pools_.empty()) {
-    createNextPool();
+  // Always start with the HTTP/3 pool if it exists.
+  ConnectionPool::Instance* pool = http3_pool_ ? http3_pool_.get() : http2_pool_.get();
+  if (!pool) {
+    pool = createNextPool();
```

Comment:
Having read through the rest of this PR, I think that we should eliminate `createNextPool()`. In this method, we have an explict `if()` down on line 320 which controls if we're doing h2 or h3. Can we simply inline the initialization of `http2_pool_` and `http3_pool_` there and eliminate `createNextPool()` since the *next* is a list operation which we really don't seem to use.

Reply:
SGTM!